From a664c8959635b854a79ccf3f8cb8599f18f963f4 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Fri, 25 Aug 2017 18:57:08 +0800
Subject: [PATCH 2462/5242] MLK-16276: PxP: Improve code compatibility

commit  0d58c7a01584c3d71978ff83d308d3bd3a0e9673 from
https://source.codeaurora.org/external/imx/linux-imx.git

g2d code has different parameter setting about stride parameter.
For compatibility with all cases of using PxP, we need add this
improved feature.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 53c8ffffec181a765f4487a9d1bf2eb05b1b78f7)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index f7b674f..9359440 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -3526,7 +3526,11 @@ static int convert_param_to_pixmap(struct pxp_pixmap *pixmap,
 	pixmap->format = param->pixel_fmt;
 	pixmap->paddr  = param->paddr;
 	pixmap->bpp    = get_bpp_from_fmt(pixmap->format);
-	pixmap->pitch  = param->width * pixmap->bpp >> 3;
+
+	if (!param->stride || (param->stride == param->width))
+		pixmap->pitch  = param->width * pixmap->bpp >> 3;
+	else
+		pixmap->pitch = param->stride;
 
 	pixmap->crop.x = param->crop.left;
 	pixmap->crop.y = param->crop.top;
-- 
1.7.9.5

